### PR TITLE
Swagger API not accept generic type TimeSeries as return type

### PR DESCRIPTION
**Pull request by thangqp**
Cause: in my opinion, the loop infinite is from the definition of type TimeSeries in recursive manner!
`TimeSeries<P extends AbstractPoint, T extends TimeSeries<P, T>>`

 
Solution: 
- Return with a parameterized type DoubleTimeSeries for timeseries and StringTimeSeries for timeline 
- Check type a service level, if KO throw an exception

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -1296,9 +1297,9 @@ public ResponseEntity<UUID> runDynamicSimulation(@Parameter(description = "study
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "All time series of dynamic simulation result"),
         @ApiResponse(responseCode = "204", description = "No dynamic simulation has been done yet"),
         @ApiResponse(responseCode = "404", description = "The dynamic simulation has not been found")})
-    public ResponseEntity<List<TimeSeries>> getDynamicSimulationTimeSeriesResult(@Parameter(description = "study UUID") @PathVariable("studyUuid") UUID studyUuid,
-                                                                       @Parameter(description = "nodeUuid") @PathVariable("nodeUuid") UUID nodeUuid) {
-        List<TimeSeries> result = studyService.getDynamicSimulationTimeSeries(nodeUuid);
+    public ResponseEntity<List<StoredDoubleTimeSeries>> getDynamicSimulationTimeSeriesResult(@Parameter(description = "study UUID") @PathVariable("studyUuid") UUID studyUuid,
+                                                                     @Parameter(description = "nodeUuid") @PathVariable("nodeUuid") UUID nodeUuid) {
+        List<StoredDoubleTimeSeries> result = studyService.getDynamicSimulationTimeSeries(nodeUuid);
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1302]: Returns `DoubleTimeSeries` instead

File: src/main/java/org/gridsuite/study/server/service/dynamicsimulation/impl/DynamicSimulationServiceImpl.java [hunk 2]
```diff
@@ -52,25 +54,47 @@ public UUID runDynamicSimulation(String receiver, UUID networkUuid, String varia
...
         UUID timeSeriesUuid = dynamicSimulationClient.getTimeSeriesResult(resultUuidOpt.get()); // get timeseries uuid
-        return timeSeriesClient.getTimeSeriesGroup(timeSeriesUuid); // get timeseries data
+
+        // get timeseries data
+        List<TimeSeries> timeSeries = timeSeriesClient.getTimeSeriesGroup(timeSeriesUuid);
+
+        // get first element to check type
+        if (timeSeries != null &&
+                !timeSeries.isEmpty() &&
+                !(timeSeries.get(0) instanceof StoredDoubleTimeSeries)) {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/study/server/service/dynamicsimulation/impl/DynamicSimulationServiceImpl.java:71]: DoubleTimeSeries

**Review by geofjamg — CHANGES_REQUESTED** [review]
@thangqp there is no problem to return an interface for swagger. We do that all the time like where returning a List. This is not the issue. Could you please post the full error stack you had.
In my opinion your issue is more about Swagger runtime not having custom our jackson serializer and so on trying to serialize to an object that has not been designed forthat.

**Comment by thangqp** [comment]
2023-02-07 10:57:50.019 ERROR 25890 --- [nio-5001-exec-6] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Handler dispatch failed; nested exception is java.lang.StackOverflowError] with root cause

java.lang.StackOverflowError: null
	at org.springframework.util.ObjectUtils.nullSafeHashCode(ObjectUtils.java:394) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.calculateHashCode(ResolvableType.java:938) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.hashCode(ResolvableType.java:934) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.util.ObjectUtils.nullSafeHashCode(ObjectUtils.java:423) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.calculateHashCode(ResolvableType.java:943) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.<init>(ResolvableType.java:148) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.forType(ResolvableType.java:1436) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.forType(ResolvableType.java:1405) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.getGenerics(ResolvableType.java:720) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.getGenerics(ResolvableType.java:724) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.core.ResolvableType.hasUnresolvableGenerics(ResolvableType.java:548) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springdoc.core.ReturnTypeParser.resolveType(ReturnTypeParser.java:85) ~[springdoc-openapi-common-1.5.13.jar:1.5.13]
	at org.springdoc.core.ReturnTypeParser.resolveType(ReturnTypeParser.java:91) ~[springdoc-openapi-common-1.5.13.jar:1.5.13]
	at org.springdoc.core.ReturnTypeParser.resolveType(ReturnTypeParser.java:91) ~[springdoc-openapi-common-1.5.13.jar:1.5.13]

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: Done

File: src/main/java/org/gridsuite/study/server/service/dynamicsimulation/impl/DynamicSimulationServiceImpl.java (hunk 2 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/study/server/service/dynamicsimulation/impl/DynamicSimulationServiceImpl.java]: Done

**Review by geofjamg — APPROVED** [review]

**Pull request merged**